src/main/java/gregtech/common/EventHandlers.java

```diff
@@ -206,9 +213,92 @@ public static void onLivingEquipmentChangeEvent(LivingEquipmentChangeEvent event
+                        walkingSpeedBonus.getWalkingSpeedBonus() != 1 &&
+                        walkingSpeedBonus.bonusSpeedCondition(player) &&
+                        walkingSpeedBonus.checkApplicableBlocks(state)) {
+                    speedBonus = walkingSpeedBonus.getWalkingSpeedBonus() - 1;
+                }
+            }
+            if (modifier != null) {
+                if (speedBonus == modifier.getAmount()) return;
+                else movementSpeed.removeModifier(BlockUtility.WALKING_SPEED_UUID);
+            } else {
+                if (speedBonus == 0) return;
+            }
+            if (speedBonus != 0) {
+                movementSpeed.applyModifier(new AttributeModifier(BlockUtility.WALKING_SPEED_UUID,
+                        "Walking Speed Bonus", speedBonus, 2));
+            }
+        }
+    }
+
+    @SuppressWarnings({ "lossy-conversions", "ConstantValue" })
+    @SubscribeEvent
+    @SideOnly(Side.CLIENT)
+    public static void onFOVUpdate(FOVUpdateEvent event) { // this event SUCKS
+        EntityPlayer player = event.getEntity();
+        IAttributeInstance movementSpeed = player.getEntityAttribute(SharedMonsterAttributes.MOVEMENT_SPEED);
+        if (movementSpeed == null || movementSpeed.getModifier(BlockUtility.WALKING_SPEED_UUID) == null) return;
+
+        float originalFov = player.capabilities.isFlying ? 1.1f : 1.0f;
+        originalFov *= (movementSpeed.getAttributeValue() / player.capabilities.getWalkSpeed() + 1) / 2;
+
+        if (player.capabilities.getWalkSpeed() == 0 || Float.isNaN(originalFov) || Float.isInfinite(originalFov)) {
+            return;
+        }
+
+        float newFov = player.capabilities.isFlying ? 1.1f : 1.0f;
+        newFov *= (computeValueWithoutWalkingSpeed(movementSpeed) / player.capabilities.getWalkSpeed() + 1) / 2;
+
+        event.setNewfov(newFov / originalFov * event.getNewfov());
+    }
+
+    /**
+     * Computes walking speed without boost from {@link BlockUtility#WALKING_SPEED_BONUS}. Skipping parent check stuff
+     * because movement speed attribute does not have any parent modifier.
+     */
+    private static double computeValueWithoutWalkingSpeed(IAttributeInstance attrib) {
+        double base = attrib.getBaseValue();
+
+        for (AttributeModifier m : attrib.getModifiersByOperation(0)) {
+            base += m.getAmount();
+        }
+
+        double applied = base;
+
+        for (AttributeModifier m : attrib.getModifiersByOperation(1)) {
+            applied += base * m.getAmount();
+        }
+
+        for (AttributeModifier m : attrib.getModifiersByOperation(2)) {
+            if (m.getID() == BlockUtility.WALKING_SPEED_UUID) continue;
+            applied *= 1 + m.getAmount();
+        }
+
+        return attrib.getAttribute().clampValue(applied);
+    }
+
+    @SubscribeEvent
+    @SideOnly(Side.CLIENT)
+    public static void onPlayerTickClient(TickEvent.PlayerTickEvent event) {
```

Comment:
Is moving this to the client event correct?

Reply:
This method was originally named `onPlayerTick`, and it had SideOnly annotation before, so I assumed it is correct.

Reply:
ok